lib/elix/responders/gif_me.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Elix.Responders.GifMe do
+  @moduledoc """
+  Searches the Giphy public API and return the URL
+  of a GIF matching the search term.
+  """
+
+  use Hedwig.Responder
+
+  @base_url "http://api.giphy.com/v1/gifs/search"
+  @api_key "dc6zaTOxFJmzC" # Giphy’s public beta API key
+  @sample_size 10
+
+  @usage """
+  gif me <term> - Replies with a GIF URL matching the term
+  """
+  hear ~r/gif me (.+)/i, msg do
+    reply(msg, get_gif_url(msg.matches[0]))
+  end
+
+  defp get_gif_url(search_term) do
+    search_term
+    |> build_api_url
+    |> make_request
+    |> get_random_image_url
+  end
```

Comment:
So something like:

```elixir
with {:ok, url}  <- build_api_url(search_term),
with {:ok, data} <- make_request(url),
     do: get_random_image_url(data)
else
  {:error, :bad_url}     -> {:error, "we couldn't process your search term"}
  {:error, :bad_request} -> {:error, "we had trouble talking to giphy"}
  {:error, :no_url}      -> {:error, "we couldn't find an image for that"}
  {:error, reason}       -> {:error, "something bad happened: #{reason}}
end
```

With the assumption that `get_random_image_url` will return `{:ok, url}`?